src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.cs

```diff
@@ -32,6 +33,17 @@ namespace System
     [Serializable]
     [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     public readonly struct DateTimeOffset : IComparable, ISpanFormattable, IComparable<DateTimeOffset>, IEquatable<DateTimeOffset>, ISerializable, IDeserializationCallback
+#if FEATURE_GENERIC_MATH
+#pragma warning disable SA1001
+        , IAdditionOperators<DateTimeOffset, TimeSpan, DateTimeOffset>,
+          IAdditiveIdentity<DateTimeOffset, TimeSpan>,
+          IComparisonOperators<DateTimeOffset, DateTimeOffset>,
+          IMinMaxValue<DateTimeOffset>,
+          ISpanParseable<DateTimeOffset>,
+          ISubtractionOperators<DateTimeOffset, TimeSpan, DateTimeOffset>,
+          ISubtractionOperators<DateTimeOffset, DateTimeOffset, TimeSpan>
```

Comment:
In practice, with such a specific combination of generic type arguments, how will this be used in practice? i.e. what's an example of a method that would successfully operate over this and not be tied directly to DateTimeOffset?

Reply:
APIs like a generic `Sum` or `Clamp` can still take advantage. Other APIs may indeed not be as usable for cases like `DateTimeOffset` or may only be usable if you can customize more than just one input type.

Reply:
I'm particularly fond of `Clamp` in this scenario.